site/en/docs/extensions/mv3/promises/index.md

```diff
@@ -0,0 +1,214 @@
+---
+layout: 'layouts/doc-post.njk'
+
+title: Using promises
+# subhead: 'How to use promises when calling extensions APIs'
+
+# This appears in the ToC of the project landing page at
+# /docs/[project-name]/. It also appears in the <meta description> used in 
+# Google Search.
+description: 'How to use promises when calling extensions APIs'
+
+# The publish date
+date: 2021-03-26
+
+# An optional updated date
+# updated: 2020-10-16
+
+---
+
+Many extension API methods support promises.  This document explains how to use promises when
+calling these methods.
+
+{% Aside "key-term" %}
+A *promise* is a JavaScript object that represents the eventual outcome of an asynchronous
+operation. For more about promises and their use, see the MDN documentation on
+[using promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises).
+{% endAside %}
+
+## Introduction
+
+Promises were introduced into Chrome not long after they were included in the ES6 specification.
+They are an important feature of modern JavaScript, providing several benefits, such as:
+
+* Streamlined error handling
+* Coding in a synchronous style for invoking asynchronous functions
+* A simple "fork and join" syntax for invoking concurrent functions
+
+Extensions can use promises beginning with Manifest V3. Many API methods support promises, and we
+are progressively adding promise support to additional API methods.
+
+{% Aside 'gotchas' %}
+Promises are not available for extensions using Manifest V2, and are not available on all API methods.
+{% endAside  %}
+
+Promises can and should be used in many circumstances. However, there are times (for example, event
+listeners) when a promise won't work and a callback is more appropriate. Methods that support
+promises also support callbacks when invoked asynchronously.
```

Comment:
```suggestion
promises also support callbacks for backwards compatibility reasons.
```

"when invoked asynchronously" – JavaScript doesn't allow you to choose whether a method is invoked synchronously or asynchronously. I'm not quite sure what this clause is attempting to address, so my current inclination is to drop it. Alternatively, we could provide more clarification on why we support both (as I've suggested here).